Sources/MongoSwift/MongoClient.swift

```diff
@@ -548,8 +548,26 @@ public class MongoClient {
         using connection: Connection? = nil,
         session: ClientSession? = nil
     ) -> EventLoopFuture<T.OperationResult> {
+        guard !self.isClosed else {
+            return self.makeFailedFuture(MongoClient.ClosedClientError)
+        }
+
+        if let session = session {
```

Comment:
I decided to add these here as it seems better to detect these cases early without having to tie up a thread in the thread pool.